lib/vrl/parser/src/lex.rs

```diff
@@ -500,10 +500,13 @@ impl<'input> Iterator for Lexer<'input> {
                     ':' => Some(Ok(self.token(start, Colon))),
                     ',' => Some(Ok(self.token(start, Comma))),
 
-                    '_' if self.test_peek(char::is_alphabetic) => {
+                    '_' if !self.test_peek(is_ident_continue) => {
+                        Some(Ok(self.token(start, Underscore)))
+                    }
+
+                    '?' if self.test_peek(char::is_alphabetic) => {
```

Comment:
I almost want to make this some sort of obscure unicode character, but `?` seems like a fairly safe bet.